### PR TITLE
commands/build: output charms to cwd when using provider (CRAFT-359)

### PR DESCRIPTION
**Pull request by cjp256**
If building in project directory, use the project path as the working
directory. The output charms will be placed in the correct directory
without needing retrieval. If outputing to a directory other than the
charm project directory, we need to output the charm outside the
project directory and can retrieve it when complete. Use the home
directory for this output inside the container.

Signed-off-by: Chris Patterson <chris.patterson@canonical.com>

### CODE REVIEW

File: charmcraft/commands/build.py [hunk 1]
```diff
@@ -254,6 +254,19 @@ def pack_charm_in_instance(
         charm_name = format_charm_file_name(
             self.metadata.name, self.config.bases[bases_index]
         )
+
+        # If building in project directory, use the project path as the working
+        # directory. The output charms will be placed in the correct directory
+        # without needing retrieval. If outputing to a directory other than the
+        # charm project directory, we need to output the charm outside the
+        # project directory and can retrieve it when complete.
+        if os.getcwd() == str(self.charmdir):
```
Comment (sergiusens) [inline charmcraft/commands/build.py:263]:
```suggestion
        cwd = pathlib.Path(os.getcwd())
        if cwd == self.charmdir.absolute():
```

File: charmcraft/commands/build.py [hunk 2]
```diff
@@ -273,14 +286,25 @@ def pack_charm_in_instance(
...
                 capture_logs_from_instance(instance)
                 raise CommandError(
                     f"Failed to build charm for bases index '{bases_index}'."
                 ) from error
 
+            if pull_charm:
+                try:
+                    instance.pull_file(
+                        source=instance_output_dir / charm_name,
+                        destination=pathlib.Path(charm_name),
```
Comment (sergiusens) [inline charmcraft/commands/build.py:301]:
```suggestion
                        destination=cwd / charm_name,
```

**Review by sergiusens — APPROVED** [review]
Looks good, just some abspath suggestions, give or take

File: charmcraft/commands/build.py (hunk 1 above)
Comment (cmatsuoka) [reply charmcraft/commands/build.py:263]: Or `pathlib.Path.cwd()`:
```
>>> import pathlib
>>> pathlib.Path.cwd()
PosixPath('/home/claudio')
```

**Review by cmatsuoka — APPROVED** [review]

File: charmcraft/commands/build.py (hunk 1 above)
Comment (sergiusens) [reply charmcraft/commands/build.py:263]: I looked for this and could not find it, thanks for the pointer :-)

File: charmcraft/commands/build.py (hunk 2 above)
Comment (cjp256) [reply charmcraft/commands/build.py]: Done!

File: charmcraft/commands/build.py (hunk 1 above)
Comment (cjp256) [reply charmcraft/commands/build.py]: Done! Thanks!

**Pull request merged**